adaptivefiltering/segmentation.py

```diff
@@ -130,124 +174,189 @@ def __init__(self, dataset=None, segmentation=None):
-            center=(self.coordinates_mean[1], self.coordinates_mean[0]),
-            crs=ipyleaflet.projections.EPSG3857,
-            scroll_wheel_zoom=True,
-            max_zoom=20,
-        )
-        self.m.add_layer(self.boundary_geoJSON)
+        :param _type: Can either be "Hillshade" or "Slope"
+        :type _type: String
 
-        # add polygon draw tool and zoom slider
-        self.add_zoom_slider()
-        self.add_polygon_control()
+        :param classification: a asprs classification that will be passed on to the visualisation function. Default = asprs[:]
+        :type classification: asprs
 
-        # setup the grid with a list of widgets
-        self.setup_grid([self.m])
+        :param resolution:resolution for the visualisation. Default = 2
+        :type resolution: int
 
-    def get_boundary(self, dataset):
-        """Takes the boundary coordinates of the  given dataset
-            through the pdal hexbin filter and returns them as a segmentation.
+        :param azimuth:azimuth for the visualisation. Default = 315
+        :type azimuth: int
 
-        :param dataset:
-            The dataset from which the map should be displayed.
-        :type dataset: Dataset
+        :param angle_altitude: angle_altitude for the visualisation. Default = 45
+        :type angle_altitude: int
 
-        :return:
-            hexbin_segmentation:
-            The Segmentation of the area from the dataset
-        :type hexbin_segmentation: Segmentation
+        :param opacity: Sets the opacity of the layer, does not trigger recalculation of layers. Default = 0.6
+        :type opacity: float
 
         """
-        from adaptivefiltering.pdal import execute_pdal_pipeline, PDALInMemoryDataSet
-
-        # convert dataset to in memory pdal dataset
-        dataset = PDALInMemoryDataSet.convert(dataset)
-
-        # execute the reprojection and hexbin filter.
-        # this is nessesary for the map to function properly.
-        hexbin_pipeline = execute_pdal_pipeline(
-            dataset=dataset,
-            config=[
-                {"type": "filters.hexbin"},
-            ],
+        from adaptivefiltering.pdal import execute_pdal_pipeline
+
+        if self.dataset == None:
+            raise Exception("No dataset was given to calculate the hillshade or slope.")
+
+        if map_type != "Hillshade" and map_type != "Slope":
+            raise Exception("map_type can only be 'Hillshade' or 'Slope'.")
+
+        # set azimuth and angle_altitude to zero for _type =="Slope"
+        # This makes it easer to find preexisting slope overlays
+        if map_type == "Slope":
+            azimuth = 0
+            angle_altitude = 0
+
+        key_from_input = (
+            "_type:"
+            + map_type
+            + ",class:"
+            + str(classification)
+            + ",res:"
+            + str(resolution)
+            + ",az:"
+            + str(azimuth)
+            + ",ang:"
+            + str(angle_altitude)
```

Comment:
Not sure a string is the best option here. If you have `**kwargs` anyway, `tuple(kwargs.items())` might do the job.